packages/agent/src/invitation/handler.ts

```diff
@@ -199,26 +202,26 @@ export const receiveCredentialFromOpenId4VciOffer = async ({
         didMethod = 'key'
       }
 
-      let key: Key | undefined = undefined
-
-      // For P-256 we first try secure enclave
-      if (keyType === KeyType.P256) {
-        key = await agent.wallet
-          .createKey({
-            keyType,
-            keyBackend: KeyBackend.SecureElement,
-          })
-          .catch((e) => {
-            agent.config.logger.warn('Could not create a key in the secure element', e as Record<string, unknown>)
-            return agent.wallet.createKey({
-              keyType,
-            })
-          })
-      } else {
-        key = await agent.wallet.createKey({
-          keyType,
-        })
-      }
+      const offeredCredentialConfiguration = supportedCredentialId
+        ? resolvedCredentialOffer.offeredCredentialConfigurations[supportedCredentialId]
+        : undefined
+
+      const shouldKeyBeHardwareBackedForMsoMdoc = false
+      //   offeredCredentialConfiguration?.format === "mso_mdoc" &&
+      //   pidSchemes?.msoMdocNamespaces.includes(
+      //     offeredCredentialConfiguration.namespace
+      //   );
+      const shouldKeyBeHardwareBackedForSdJwtVc =
+        offeredCredentialConfiguration?.format === 'vc+sd-jwt' &&
+        pidSchemes?.sdJwtVcVcts.includes(offeredCredentialConfiguration.vct)
+
+      // TODO: add mso-mdoc config from above
+      const shouldKeyBeHardwareBacked = shouldKeyBeHardwareBackedForSdJwtVc ?? shouldKeyBeHardwareBackedForMsoMdoc
```

Comment:
?? will only go to the next one if the value is undefined, but shouldKeyBeHardwareBackedForSdJwtVC will always have a value. I think you mean || in this case?